From 43a3011cf7edad94a2b3d69efb21b45fe018c9d1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:54 -0400
Subject: [tools] perf/hists: Free srcline when freeing hist_entry

Message-id: <1408462094-14194-126-git-send-email-jolsa@redhat.com>
Patchwork-id: 88128
O-Subject: [PATCH RHEL7.1 BZ1131394 125/325] perf hists: Free srcline when freeing hist_entry
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 909b143162de7af310d2a9351220030260ebe728
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Wed Sep 11 14:09:27 2013 +0900

description
===========
We've been leaked srcline of hist_entry, it should be freed also.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index f3278a3..e6fc38a 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -535,6 +535,7 @@ void hist_entry__free(struct hist_entry *he)
 {
  free(he->branch_info);
  free(he->mem_info);
+ free(he->srcline);
  free(he);
 }
 
-- 
1.7.1